pkg/api/documentsschema/documentsschema.proto

```diff
@@ -0,0 +1,252 @@
+syntax = "proto3";
+
+package immudb.documentsschema;
+
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/struct.proto";
+import "schema.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+
+option go_package = "github.com/codenotary/immudb/pkg/api/documentsschema";
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+  base_path: "/api/v2",
+  info: {
+    title: "immudb REST API v2";
+    description: "Objects storage API"
+  };
+  security_definitions: {
+    security: {
+      key: "bearer"
+      value: {
+        type: TYPE_API_KEY
+        in: IN_HEADER
+        name: "grpc-metadata-sessionid"
+        description: "Session token: <token>"
+      }
+    }
+  }
+  security: {
+    security_requirement: {
+      key: "bearer"
+    }
+  }
+};
+
+message DocumentInsertRequest {
+  string collection = 1;
+  repeated google.protobuf.Struct document = 2;
+}
+
+message DocumentInsertResponse {
+  schema.VerifiableTx proof = 1;
+}
+
+
+enum QueryOperator {
+  EQ = 0;
+  GT = 1;
+  GTE = 2;
+  LT = 3;
+  LTE = 4;
+  LIKE = 5;
+}
+
+message IndexValue {
+  oneof kind {
+    google.protobuf.NullValue null_value = 1;
+    double number_value = 2;
+    string string_value = 3;
+    bool bool_value = 4;
+  }
+}
+
+message DocumentQuery {
+  string field = 1;
+  QueryOperator operator = 2;
+  google.protobuf.Value value = 3;
+}
+
+
+message DocumentSearchRequest {
+  string collection = 1;
+  repeated DocumentQuery query = 2;
+  uint32 page = 3;
+  uint32 perPage = 4;
+}
+
+message DocumentSearchResponse {
+  repeated google.protobuf.Struct results = 1;
+  uint32 page = 2;
+  uint32 perPage = 3;
+  uint32 entriesLeft = 4;
```

Comment:
to be aware of: determining `entriesLeft` may be hard to achieve...

Reply:
is it worth to include page and perPage in the response?

Reply:
Yes and no again

in terms of simplicity yes, because when you are doing for example N requests at once for example with multiprocessing pool or multithreading pool - the order of returned response is not guaranted
so when perpage and page is not returned with response it will force user to add it these fields himself to distinguish fields

or for example with
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all
Promise all will just return value of responses
Again - force user to either include values into reponse, or wrap reponse in something

Not big deal here, we can remove, but i dont see minuses from that